server/src/modules/post/post.service.ts

```diff
@@ -212,31 +212,15 @@ export class PostService {
     description: string
     userId: string
     tagname: string[]
-  }): Promise<HelperResult> => {
+  }): Promise<string> => {
     const tagList = await this.getExistingTagsFromRequestTags(newPost.tagname)
 
     if (newPost.tagname.length !== tagList.length) {
-      return [
-        helperFunction.responseHandler(
-          false,
-          400,
-          'At least one tag does not exist',
-          null,
-        ),
-        null,
-      ]
+      return Promise.reject('At least one tag does not exist')
```

Comment:
As discussed, we should favour throwing Errors instead of using `Promise.reject()`